sway-core/src/semantic_analysis/ast_node/declaration/impl_trait.rs

```diff
@@ -257,6 +257,11 @@ impl TypedImplTrait {
                     kind,
                     ..
                 }) => matches!(kind, sway_ast::intrinsics::Intrinsic::GetStorageKey),
+                TypedExpressionVariant::Reassignment(reassignment) => {
+                    expr_contains_get_storage_index(&reassignment.rhs)
+                }
+                // FIXME: shouldn't this check the rhs?
```

Comment:
This is in `fn expr_contains_get_storage_index`. Do we need to check the rhs of a storage assignment for `GetStorageKey`?

Reply:
Yes, since it could contain any expression (as per the AST definition) including a `get_storage_index`.